src/NTestController/NTestController.cs

```diff
@@ -56,16 +58,37 @@ public static void Main(string[] args)
                     // Load plugins.
                     Dictionary<PluginType, IPlugin> plugins = GetPlugins(_options.ConfigFile);
 
+                    var xmlDoc = XmlUtils.LoadXmlDocument(_options.ConfigFile);
+                    var defaultsNode = GetDefaultsXmlNode(xmlDoc);
+                    var platforms = GetPlatforms(xmlDoc, defaultsNode);
+
                     // Execute Reader plugin.
-                    plugins[PluginType.TestReader].Execute();
+                    var readerPlugin = plugins[PluginType.TestReader] as IReaderPlugin;

```

Comment:
Should check if each of these plugins exist in the Dictionary before trying to do anything with them...